### PR TITLE
Update modular imports section of javascript API category

### PR DESCRIPTION
**Pull request by Amplifiyer**
*Issue #, if available:*

*Description of changes:* With modularization update, API category is now split between Rest and Graphql modules while also providing backwards compatible API module.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: js/api.md [hunk 1]
```diff
@@ -2212,18 +2212,29 @@ postData();
 
 ## Using Modular Imports
 
-If you only need to use API, you can run: `npm install @aws-amplify/api` which will only install the API module.
-Note: if you're using Cognito Federated Identity Pool to get AWS credentials, please also install `@aws-amplify/auth`.
-Note: if you're using Graphql, please also install `@aws-amplify/pubsub`
+Amplify V3 now supports Rest and Graphql modular imports such as if you only need to use Rest API,
```
Comment (ericclemmons) [inline js/api.md]:
```suggestion
Amplify V3 now supports REST and GraphQL modular imports such as if you only need to use REST API,
```

File: js/api.md [hunk 2]
```diff
@@ -2212,18 +2212,29 @@ postData();
...
+you can run: `npm install @aws-amplify/api-rest` which will only install the Rest API module and
```
Comment (ericclemmons) [inline js/api.md]:
```suggestion
you can run: `npm install @aws-amplify/api-rest` which will only install the REST API module and
```

File: js/api.md [hunk 3]
```diff
@@ -2212,18 +2212,29 @@ postData();
...
 
-API.configure();
+```javascript
+import GraphqlAPI, { graphqlOperation } from '@aws-amplify/api-graphql';
+ 
+GraphqlAPI.configure();
 ```
 
+Note: if you're using Cognito Federated Identity Pool to get AWS credentials, please also install `@aws-amplify/auth`.
+Note: if you're using Graphql, please also install `@aws-amplify/pubsub`
```
Comment (ericclemmons) [inline js/api.md]:
```suggestion
Note: if you're using GraphQL, please also install `@aws-amplify/pubsub`
```

File: js/api.md [hunk 4]
```diff
@@ -2212,18 +2212,29 @@ postData();
...
+Then in your code, you can import this module by:
 
 ```javascript
-import API, { graphqlOperation } from '@aws-amplify/api';
+import RestAPI from '@aws-amplify/api-rest';
+
+RestAPI.configure();
+```
+
+Similarly to use only Graphql API you can run: `npm install @aws-amplify/api-graphql` and then in your code
```
Comment (ericclemmons) [inline js/api.md]:
```suggestion
Similarly to use only the Graphql API you can run: `npm install @aws-amplify/api-graphql` and then in your code:
```

**Comment by Amplifiyer** [comment]
> Can you share a quip doc, or GitHub issue that has more information on the challenges that led to this? I've validated that the `modularization` branch saves **hundreds** of KB, but wanted to understand this specific split better for myself.

This specific split is to save bundle size when customers are only using REST since the size of graphql module is significant. In the original RFC, there has been a recent comment regarding this as well https://github.com/aws-amplify/amplify-js/issues/3365#issuecomment-598266678

File: js/api.md [hunk 5]
```diff
@@ -2212,18 +2212,29 @@ postData();
 
 ## Using Modular Imports
 
-If you only need to use API, you can run: `npm install @aws-amplify/api` which will only install the API module.
-Note: if you're using Cognito Federated Identity Pool to get AWS credentials, please also install `@aws-amplify/auth`.
-Note: if you're using Graphql, please also install `@aws-amplify/pubsub`
+Amplify V3 now supports REST and GraphQL modular imports such as if you only need to use REST API,
+you can run: `npm install @aws-amplify/api-rest` which will only install the REST API module and
+reduce you application bundle size by not including any graphql related modules.
```
Comment (sammartinez) [inline js/api.md]:
```suggestion
reduce your application bundle size by not including any GraphQL related modules.
```

File: js/api.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Amplifiyer) [reply js/api.md]: i think it was already fixed by previous suggestion from Eric.

File: js/api.md [hunk 6]
```diff
@@ -2212,18 +2212,29 @@ postData();
...
+import RestAPI from '@aws-amplify/api-rest';
+
+RestAPI.configure();
+```
+
+Similarly to use only the Graphql API you can run: `npm install @aws-amplify/api-graphql` and then in your code:
 
-API.configure();
+```javascript
+import GraphqlAPI, { graphqlOperation } from '@aws-amplify/api-graphql';
```
[the remark this answers is not in the document]
Comment (Amplifiyer) [reply js/api.md]: I don't mind either way, these are default exports so users can call it whatever they want.  Named export however is actually `GraphQLAPI` Should I change it for this example here as well?

[the remark this answers is not in the document]
Comment (sammartinez) [reply js/api.md]: Yeah might as well so we can have naming conventions be correct for customers and what theyre used to.

[the remark this answers is not in the document]
Comment (ericclemmons) [reply js/api.md]: @Amplifiyer's right, this is a cosmetic change.

I checked the original issue and `GraphQLAPI` + `RestAPI` were mentioned there:
> https://github.com/aws-amplify/amplify-js/issues/3365#issuecomment-502009495

File: js/api.md [hunk 7]
```diff
@@ -2212,18 +2212,29 @@ postData();
...
+import RestAPI from '@aws-amplify/api-rest';
+
+RestAPI.configure();
+```
+
+Similarly to use only the GraphQL API you can run: `npm install @aws-amplify/api-graphql` and then in your code:
 
-API.configure();
+```javascript
+import GraphqlAPI, { graphqlOperation } from '@aws-amplify/api-graphql';
```
Comment (ericclemmons) [inline js/api.md]: Suggestion to match https://github.com/aws-amplify/amplify-js/issues/3365#issuecomment-502009495:

```suggestion
import GraphQLAPI, { graphqlOperation } from '@aws-amplify/api-graphql';
```

File: js/api.md [hunk 8]
```diff
@@ -2212,18 +2212,29 @@ postData();
...
+ 
+GraphqlAPI.configure();
```
Comment (ericclemmons) [inline js/api.md]: Suggestion to match https://github.com/aws-amplify/amplify-js/issues/3365#issuecomment-502009495:

```suggestion
GraphQLAPI.configure();
```

**Pull request merged**